Symfony/CS/Fixer/Contrib/OldStyleConstructorFixer.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/*
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Fixer\Contrib;
+
+use Symfony\CS\AbstractFixer;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * @author Szymon Piasecki <szymon.piasecki@gmail.com>
+ */
+class OldStyleConstructorFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, $content)
+    {
+        $tokens = Tokens::fromCode($content);
+
+        // As of PHP 5.3.3, methods with the same name as the last element of a namespaced class name will no longer be treated as constructor.
+        // This change doesn't affect non-namespaced classes.
+        $classHasNamespace = $tokens->findGivenKind(T_NAMESPACE);
```

Comment:
``$classHasNamespace`` is a weird name. the variable name let me think it will contain a boolean value. But the fact that the class has a namespace or no is actually handled by ``!empty($classHasNamespace)``.